frontend/src/components/characters/Characters.jsx

```diff
@@ -1,47 +1,122 @@
 import { Card } from "./subcomponents/Card";
 import React, { useState, useEffect } from "react";
 import { useNavigate } from "react-router-dom";
-import { useApi, useAuthentication } from "../../hooks";
-import { ClipLoader } from "react-spinners";
+import useApi from "../../hooks/useApi";
+import { FilterByCategory } from "./subcomponents/FilterByCategory";
+import { SearchBar } from "./subcomponents/SearchBar";
 
-export default function Characters({ ownedOnly }) {
-  const { isLoggedIn } = useAuthentication();
-  const [loggedIn, setLoggedIn] = useState(null);
+export default function Characters({ myCharacters }) {
   const navigate = useNavigate();
-  const [characters, setCharacters] = useState([]);
   const { get } = useApi();
+
+  const [characters, setCharacters] = useState([]);
+  const [races, setRaces] = useState([]);
+  const [classes, setClasses] = useState([]);
+  const [url, setUrl] = useState("characters");
+
+  const [raceParams, setRaceParams] = useState([]);
+  const [classParams, setClassParams] = useState([]);
+
+  const [detectClick, setDetectClick] = useState([]);
+
   useEffect(() => {
-    if (ownedOnly && loggedIn === false) {
-      navigate("/");
```

Comment:
Er wordt hier niet meer gecheckt of je wel op deze pagina mag zijn, request params hoor je mee te geven als object via de daarvoor gemaakte method parameter ( get(url, params),  ex: get('characters', { race: "dwarf", class: "paladin" } ), axios zet dit om naar request params